.gitignore

```diff
@@ -1,3 +1,17 @@
+# Temp files
+*~
+*.bak
+*.backup
+\#*
```

Comment:
Yes, Emacs use (by default) this pattern for auto-save files. https://www.emacswiki.org/emacs/BackupDirectory (see "Delete auto-save files")